### PR TITLE
Fix OVAL for audit_rules_unsuccessful_file_modification_*

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

The OVAL template was too broad and it allowed key=anything at the end of the audit rules. The lines mentioned in the rule description were not in sync with the actual file which is used for testing. The testing file contains
key=unsuccessful-access
but the rule description contained
key=access
Looking into example files provided by Audit I believe key=unsuccessful-access is correct.
@stevegrubb could you please confirm that?
If it is correct there are more rule descriptions need fixing.

#### Rationale:

Tests for the rule were not passing.

### CODE REVIEW

File: shared/templates/template_OVAL_audit_rules_unsuccessful_file_modification [hunk 1]
```diff
@@ -52,7 +52,7 @@
       <value>^[\s]*-a[\s]+always,exit[\s]+(?:-F[\s]+arch=b64[\s]+)(?:.*(-S[\s]+{{{ NAME }}}[\s]+|([\s]+|[,]){{{ NAME }}}([\s]+|[,]))).*</value>
   </constant_variable>
   <constant_variable id="var_arufm_{{{ NAME }}}_tail" version="1" datatype="string" comment="audit rule auid and key">
-    <value>[\s]+(?:-F\s+auid>={{{ auid }}}[\s]+)(?:-F\s+auid!=(unset|4294967295)[\s]+)(?:-k[\s]+|-F[\s]+key=)[\S]+[\s]*$</value>
+    <value>[\s]+(?:-F\s+auid>={{{ auid }}}[\s]+)(?:-F\s+auid!=(unset|4294967295)[\s]+)(?:-k[\s]+|-F[\s]+key=)unsuccesful-access[\s]*$</value>
```
Comment (yuumasato) [inline shared/templates/template_OVAL_audit_rules_unsuccessful_file_modification]: This hardcode key won't align with all rules that use this template, see for example these other rules:
- delete: [audit_rules_unsuccessful_file_modification_unlink](https://github.com/ComplianceAsCode/content/blob/f42bc985ea3ec1704ae561634baa2d4e5920601e/linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_unlink/rule.yml)
- permission change: [audit_rules_unsuccessful_file_modification_fchown](https://github.com/ComplianceAsCode/content/blob/f42bc985ea3ec1704ae561634baa2d4e5920601e/linux_os/guide/system/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification_fchown/rule.yml)

File: shared/templates/template_OVAL_audit_rules_unsuccessful_file_modification_o_creat [hunk 2]
```diff
@@ -54,7 +54,7 @@
       <value>^[\s]*-a[\s]+always,exit[\s]+(?:-F[\s]+arch=b64[\s]+)(?:-S[\s]+(?:[\S]+,)*({{{ SYSCALL }}})(?:,[\S]+)*)[\s]+</value>
   </constant_variable>
   <constant_variable id="var_audit_rule_{{{ SYSCALL }}}_o_creat_tail" version="1" datatype="string" comment="audit rule auid and key">
-    <value>[\s]+(?:-F\s+auid>={{{ auid }}}[\s]+)(?:-F\s+auid!=(unset|4294967295)[\s]+)(?:-k[\s]+|-F[\s]+key=)[\S]+[\s]*$</value>
+    <value>[\s]+(?:-F\s+auid>={{{ auid }}}[\s]+)(?:-F\s+auid!=(unset|4294967295)[\s]+)(?:-k[\s]+|-F[\s]+key=)unsuccesful-create[\s]*$</value>
```
Comment (yuumasato) [inline shared/templates/template_OVAL_audit_rules_unsuccessful_file_modification_o_creat]: The hard coded key is aligned with the rules that use it.
But people may want to use different keys, see for example: https://github.com/ComplianceAsCode/content/pull/2715

AFAIK, the keys are there for usability reasons: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/security_guide/sec-defining_audit_rules_and_controls#bh-defining_file_rules_with_auditctl
> key_name is an optional string that helps you identify which rule or a set of rules generated a particular log entry.

**Comment by vojtapolasek** [comment]
@yuumasato  Thanks for clarification. Closing this PR and proposing fix in a new PR.

**Comment by stevegrubb** [comment]
The correct key is unsuccessful-access. But I guess this doesn't matter now.

**Comment by yuumasato** [comment]
> The correct key is unsuccessful-access.

@stevegrubb Can you elaborate more on that?
Is my understand correct that the `key` as a note to your future self when going through the events?

**Comment by stevegrubb** [comment]
The key is a tag added to an event that tells you why the event happened. For example, you might want to see what kinds of events have occurred based on the audit rules as opposed to hard coded events. You would run:
aureport --summary --key

Then based on this, you may decide a key looks interesting and see who's causing it: 
ausearch --key unsuccessful-access --raw | aureport --summary --user

Maybe a user then catches your eye and you want to see what command they run:
ausearch --key unsuccessful-access --auid john --raw | aureport --summary --command

The key field anchors the ability to do searches to see who's tripping across the rules.

**Pull request closed without merging**